### DISCUSSION TITLE
vs code as a sidecar container

### DISCUSSION THREAD

**Discussion by iham** [coder/code-server#2493] [Q&A] [answered]
running code-server as a sidecar container to another container and use the [remote container(s)](https://code.visualstudio.com/docs/remote/containers) would be awesome!

![Screenshot 2020-12-19 at 18 52 22](https://user-images.githubusercontent.com/36721/102695993-57dc1480-422b-11eb-911d-30ec644eeb86.png)

but "local os" is just another container in your cluster or stack - k8s or swarm or whatever.

imagine using vs code via your browser (instead of local) as a service next to your app(s) and step into a coding session on your app (python, node, go, whatever). with only the extensions it needs there, code-completion and terminal tools you want in that context; but nothing more.

is that possible and i don't get the docs or is this beyond the powers of vs code/remote devel/code server?

to explain again:
i don't wan't to have a code-server and code INSIDE of it. i search for how to run vs code in the browser and connect to a container NEXT to it. as mentioned before: like remote dev, but no local installation of vs code as this should run on a server already and be accessed via browser.

1. kubernets pod (python app)
    1. python app container with specifics installed (pip)
    2. vs code with docker extensions allowed to remotly code inside the python container
    3. maybe a logger or something else
2. kubernetes pod b (node app)
    1. node app container with specifics installed (yarn)
    2. vs code with docker extensions allowed to remotly code inside the node container 
    3. maybe a logger or something else

both app containers should be agnostic about vs-code, so not caring about it at all.

is that even possible or am i completely thinking the wrong direction?

**Comment by bpmct** [comment, marked as answer]
This would be very awesome! If I'm understanding correctly, #1315 should be all we would need to implement this.

**Reply by iham** [reply]
thank you for pointing me into that direction - i can see the "problems" now - licenses ... "love" em.
